### ISSUE TITLE
Audio Test crashes Post install

### ISSUE THREAD

**Issue by ppumkin** [trueos/trueos-core#111]
`Hey.. you kept on nagging us to try out TrueOS on BSDNow... Here is me trying it out :)`

Sorry for really bad reproduction steps.

When I eventually got into post installation (TrueOS first-time setup wizard) when I clicked on Test on the Audio page, the wizard froze up. Could not click next or anything. After trying to switch consoles, i eventually could but the wizard was still frozen.. Had to reboot, and redo the wizard from scratch.

Using `pciconf -lvc` i found the audio hardware it "detected"?
(sorry, another screenshot. I know how to switch consoles using [ctrl+alt Fx] but when I selected text in one console I could not paste it into the console running X into a text document?)

![Imgur](http://i.imgur.com/lDPRJ8Eh.jpg)

Second time round I didn't touch nu'ting man... I eventaully got into Desktop and et'viola. Sound worked!? I got the nice welcome chime.

**Comment by ppumkin** [comment]
Good *****
Turn down the volume before doing that! 😆 

I run that from the DE console and yes, there was static. The issue occurred during the post installation screen, when I pressed test. It literally froze everything. What is behind that button? Can I run that post installation test again?

After it rebooted (hard cycle because of reboot bug) the audio worked fine on DE. I wonder if it is some kind of auto detect bug???

**Comment by pkgdemon** [comment]
@beanpole135  do you have any thoughts as to why the QA sound test application crashes?

**Comment by beanpole135** [comment]
No clue here. The audio test in the firstboot wizard is almost identical to the Lumina startup chimes - it just uses a different audio file for playback. I have seen a number of warnings/errors about hdac devices on the boot console of some systems, perhaps there is something in that driver which takes some time to settle down before it can be used?

**Comment by ppumkin** [comment]
@beanpole135  @pkgdemon I would be more than happy to try and do it again and capture some logging somehow. Even if it means reinstalling from scratch and then attaching some kind of log dump before I press that button (if it reproduces- it may as well been a one off??) 

Maybe its just an assumption that is false and causing some catastrophic fail - Freezing up is pretty catastrophic when pressing test button and not a very good experience. Atleast rebooting allowed me to go through the process again.. but I had to redo everything from step 1.

Let me know what I can do. I am pretty competent with development and debugging (albeit I am a .NET dev ... but I have taken the dive in the BSD pool - I may as well try and help out instead of complain)

**Comment by pkgdemon** [comment]
@ppumkin Sorry for the late response.  I was able to reproduce this on another system.  I can hang the GUI on a system where sound is unsupported by default.  Just by clicking the sound test a few times when it is not working.

I have found a workaround to fix my sound as well.  Which may fix yours:

``` sysctl dev.hdac.0.polling=1"```

We have added a post installation option after the sound wizard to "Enable HDAC polling".  It will basically just add ```dev.hdac.0.polling=1"``` to ```/etc/sysctl.conf```.  This fixes sound on some intel devices.

Let me know if this works for you.
